app/src/main/java/id/rizmaulana/covid19/data/mapper/CovidDataMapper.kt

```diff
@@ -20,25 +20,24 @@ object CovidDailyDataMapper {
             response.incrementRecovered,
             response.incrementConfirmed
         )
-    }
+    } ?: emptyList()
 }
 
 object CovidOverviewDataMapper {
 
-    fun transform(response: CovidOverview?) = if(response != null) listOf(OverviewItem(
-        response.confirmed?.value ?: 0,
-        response.recovered?.value ?: 0,
-        response.deaths?.value ?: 0))
-    else emptyList()
+    fun transform(response: CovidOverview?) = OverviewItem(
+        response?.confirmed?.value ?: 0,
+        response?.recovered?.value ?: 0,
+        response?.deaths?.value ?: 0)
 }
 
 object CovidPinnedDataMapper {
 
-    fun transform(response: CovidDetail?) = if(response != null) listOf(PinnedItem(
+    fun transform(response: CovidDetail?) : PinnedItem? = if(response != null) PinnedItem(
```

Comment:
Or you can do this way

```
fun transform(response: CovidDetail?) = reponse?.map{
   PinnedItem(
      it.confirmed,
      ....
   )
}
```

Reply:
Map unresolved @rizmaulana, I think because it is not a collection

Reply:
Oh i see